fem/coefficient.hpp

```diff
@@ -308,16 +308,16 @@ class VectorArrayCoefficient : public VectorCoefficient
    explicit VectorArrayCoefficient(int dim);
 
    /// Returns i'th coefficient.
-   Coefficient &GetCoeff(int i) { return *Coeff[i]; }
+   Coefficient* GetCoeff(int i) { return Coeff[i]; }
```

Comment:
I'd like to keep the interface unchanged - it is easier for existing users to upgrade.

Reply:
I tend to agree...  maybe we can have another method, `GetCoeffPtr` that returns a pointer? The rest of the PR looks good to me.

Reply:
I understand you want to keep the current interface, but I still think that returning a reference is non-sense in this case; also, since the functionality in this PR allows for null entries in the coefficient arrays, returning a reference of any of these allowed nulls isn't undefined behaviour?
For users to upgrade, a compilation error with the new interface is better than undefined behaviour at runtime.

Can you give me an example where the return by reference case can be useful?

Reply:
I searched for usages of `GetCoeff` and couldn't find any in our codes, so I'm actually OK to break compatibility in this particular case.

Reply:
I agree that returning a pointer is better choice with the special treatment of the `NULL` case, if there was no method with the same name already. However, in the current situation I'd prefer if we call the new method something else. I think we can add `deprecated` (in the doxygen comment) to the old method and point to the new one.

That said, the above would be my preferred solution but it is not a "line in the sand". :-)

@tzanio, what do you think?

Reply:
We all agree that the pointer versions are the better technical decision, so I vote to that and documenting the API change in the `CHANGELOG` file as we've done in the past: https://github.com/mfem/mfem/blob/master/CHANGELOG#L353.

Reply:
OK. Let's do that.